File: examples/GenerateFile.m
% This does exactly the same as GenerateFile.c or GenerateFile.py

DIM = 100;
NPOINTS = 1000;

f = netcdf('testfile.nc','c');
f.doc = 'test file for ncDataReader2, created by GenerateFile.m';
f.version = '2.2.0';
f.foo = 42.42;
f.bar = 1234;

% create dimension and variables in 1D
f('time') = DIM;

f{'time'} = ncdouble('time');
f{'time'}.scale_factor = 2.5;
f{'time'}.add_offset = 0.0;
f{'time'}.extrapolation = 'periodic';

f{'test1D'} = ncdouble('time');
f{'test1D'}.scale_factor = 2.0;
f{'test1D'}.add_offset = 3.0;
f{'test1D'}.interpolation = 'akima';
f{'test1D'}.smoothing = 500.0;

step = 2.0 * pi / (DIM - 1.0);
for i = 1:1:DIM
    x = pi + i*step + 0.3*rand()*step;
    f{'time'}(i) = x;
    f{'test1D'}(i) = sin(x) + 0.1*rand();
end

% create dimension and variables for 2D
f('npoints') = NPOINTS;
f('xyz') = 3;

f{'points'} = ncdouble('npoints', 'xyz');

for i = 1:1:NPOINTS
    x = 10.0 * rand() - 5.0;
    y = 10.0 * rand() - 5.0;
    f{'points'}(i,1) = x;
    f{'points'}(i,2) = y;
    f{'points'}(i,3) = tanh(x*y);
end

close(f)

File: NcDataReader2/Resources/examples/GenerateFile.py
#!/usr/bin/env python
# This does exactly the same as GenerateFile.c .
# There are several netCDF interfaces for python available,
# this script will try use the one from scipy and fall back to pupynere.
# (http://pypi.python.org/pypi/pupynere/).

try:
    from scipy.io import *
except:
    from pupynere import *
    
from math import pi, sin, tanh
from random import random

DIM = 100
NPOINTS = 1000

def createFile(fileName):
    f = netcdf_file(fileName, 'w')
    f.doc = 'test file for ncDataReader2, created by GenerateFile.py'
    f.version = '2.3.0'
    f.foo = 42.42
    f.bar = 1234
    return f

def create1DVar(f):
    f.createDimension('time', DIM)
    time = f.createVariable('time', 'd', ('time', ))
    time.scale_factor = 2.5
    time.add_offset = 0.0
    time.extrapolation = 'periodic'

    test = f.createVariable('test1D', 'd', ('time', ))
    test.scale_factor = 2.0
    test.add_offset = 3.0
    test.interpolation = 'akima'
    test.smoothing = 500.0

    step = 2.0 * pi / (DIM - 1.0)
    for i in range(DIM):
        x = pi + i*step + 0.3*random()*step
        time[i] = x
        test[i] = sin(x) + 0.1*random()

def create2DVar(f):
    f.createDimension('npoints', NPOINTS)
    f.createDimension('xyz', 3)
    points = f.createVariable('points', 'd', ('npoints', 'xyz'))
    for i in range(NPOINTS):
        x = 10.0 * random() - 5.0
        y = 10.0 * random() - 5.0
        points[i] = (x, y, tanh(x*y))

if __name__ == '__main__':
    import sys
    if len(sys.argv) > 1:
        f = createFile(sys.argv[1])
    else:
        f = createFile('testfile.nc')
    create1DVar(f)
    create2DVar(f)
    f.sync()
    f.close()
